compiler-rt/lib/sanitizer_common/sanitizer_fuchsia.cpp

```diff
@@ -243,6 +290,12 @@ void UnmapOrDieVmar(void *addr, uptr size, zx_handle_t target_vmar) {
 
   zx_status_t status =
       _zx_vmar_unmap(target_vmar, reinterpret_cast<uintptr_t>(addr), size);
+  if (status != ZX_OK && target_vmar == gSanitizerHeapVmar) {
```

Comment:
Check for the specific error code returned for this case, which is `ZX_ERR_INVALID_ARGS`.